File: Homework/Homework_3/zhou-ke-xuan/learn/detect.py
import cv2

# 读取图像
image = cv2.imread('armor.png')

# 检查图像是否成功读取
if image is None:
    print("图像读取失败，请检查文件路径")
else:
    # 将图像转换为灰度图像
    gray_image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)

    # 显示原始图像和灰度图像
    cv2.imshow('Original Image', image)
    cv2.imshow('Gray Image', gray_image)

    # 等待按键按下，然后关闭所有窗口
    cv2.waitKey(0)
    cv2.destroyAllWindows()

    # 保存灰度图像
    cv2.imwrite('armor_gray.png', gray_image)

File: Homework/Homework_3/zhang-guo-yuan/test 1/detect.cpp
#include<iostream>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>
#include<opencv2/imgproc.hpp>
int main(){
    cv::Mat image = cv::imread("../armor.png");

    if(image.empty()){
        std::cout<<"can't read the picture,detect the way of it";
        return -1;
    }
    cv::Mat gray_image;
    cv::cvtColor(image , gray_image,cv::COLOR_BGR2GRAY);
    cv::imshow("Original Image" , image);
    cv::imshow("Gray Image" , gray_image);
    cv::waitKey(0);
    cv::destroyAllWindows();
    cv::imwrite("armor_gray.png" , gray_image);
    return 0;
}
